crates/sqlexec/src/extension_codec.rs

```diff
@@ -991,3 +1141,268 @@ impl<'a> PhysicalExtensionCodec for GlareDBExtensionCodec<'a> {
+            )
+        })?;
+
+    let window_function: WindowFunction = proto
+        .window_function
+        .as_ref()
+        .ok_or_else(|| DataFusionError::Internal("missing window function".to_string()))?
+        .try_into()?;
+
+    create_window_expr(
+        &window_function,
+        proto.name.clone(),
+        &window_node_expr,
+        &partition_by,
+        &order_by,
+        Arc::new(window_frame),
+        input_schema,
+    )
+}
+
+fn try_from_window_expr(
+    window_expr: Arc<dyn WindowExpr>,
+) -> Result<protogen::sqlexec::physical_plan::GlarePhysicalWindowExprNode> {
+    let expr = window_expr.as_any();
+
+    let mut args = window_expr.expressions().to_vec();
+
+    let window_function = if let Some(built_in_window_expr) =
+        expr.downcast_ref::<BuiltInWindowExpr>()
+    {
+        let expr = built_in_window_expr.get_built_in_func_expr();
+        let built_in_fn_expr = expr.as_any();
+
+        let builtin_fn = if built_in_fn_expr.downcast_ref::<RowNumber>().is_some() {
+            datafusion_proto::protobuf::BuiltInWindowFunction::RowNumber
+        } else if let Some(rank_expr) = built_in_fn_expr.downcast_ref::<Rank>() {
+            match rank_expr.get_type() {
+                RankType::Basic => datafusion_proto::protobuf::BuiltInWindowFunction::Rank,
+                RankType::Dense => datafusion_proto::protobuf::BuiltInWindowFunction::DenseRank,
+                RankType::Percent => datafusion_proto::protobuf::BuiltInWindowFunction::PercentRank,
+            }
+        } else if built_in_fn_expr.downcast_ref::<CumeDist>().is_some() {
+            datafusion_proto::protobuf::BuiltInWindowFunction::CumeDist
+        } else if let Some(_ntile_expr) = built_in_fn_expr.downcast_ref::<Ntile>() {
+            args.insert(
+                0,
+                Arc::new(Literal::new(datafusion::common::ScalarValue::Int64(Some(
+                    0, // Can't get "n" without public fn
+                )))),
+            );
+            datafusion_proto::protobuf::BuiltInWindowFunction::Ntile
+        } else if let Some(window_shift_expr) = built_in_fn_expr.downcast_ref::<WindowShift>() {
+            args.insert(
+                1,
+                Arc::new(Literal::new(datafusion::common::ScalarValue::Int64(Some(
+                    window_shift_expr.get_shift_offset(),
+                )))),
+            );
+            // if let Some(default_value) = window_shift_expr.get_default_value() {
+            //     args.insert(2, Arc::new(Literal::new(default_value)));
+            // }
+            if window_shift_expr.get_shift_offset() >= 0 {
+                datafusion_proto::protobuf::BuiltInWindowFunction::Lag
+            } else {
+                datafusion_proto::protobuf::BuiltInWindowFunction::Lead
+            }
+        } else if let Some(_nth_value_expr) = built_in_fn_expr.downcast_ref::<NthValue>() {
+            // match nth_value_expr.get_kind() {
```

Comment:
why comment this rather than delete it?

Reply:
This PR is me messing around with stuff from the mentioned DF PR. Wanted to test it out, but there are things that aren't public which I just commented out for now.